include/seastar/core/circular_buffer.hh

```diff
@@ -26,6 +26,7 @@
 #include <seastar/util/concepts.hh>
 #include <memory>
 #include <algorithm>
+#include <type_traits>
```

Comment:
This patch doesn't belong to "io_request: <something>" set

Reply:
are you referring 

- 96ff6ff041c43009874d62317a45f9a6916f7478 (core/circular_buffer: allow `const` specifier in value_type) ? if that's the case. and
- 64e404b0c6433e90b774fb54b41d971d21108fe8 (core/circular_buffer: refactor loop in circular_buffer::erase())

? 

they are included here just because of https://github.com/scylladb/seastar/pull/1250#issuecomment-1295139142 . so i can assure that the first commit would not prevent us from having `const` semantics of `io_request`.

i'd be happy to move the `core/circular_buffer` changes into a follow up change once we are fine with them. what do you think?

Reply:
Ah, I see. OK, making circular_buffer work with consts is fine to have here. Refactoring -- I'd prefer seeing it as separate PR (unless I'm missing something again and it's also needed to make const-ification compile)

Reply:
no, the loop change is not a necessity for adding `const`. but that change has a tiny overlap with  `core/circular_buffer: allow `const` specifier in value_type` change. so it would be very likely it will conflict with this PR. anyway, i agree with you that, strictly speaking, it should belong to another PR. just extracted it into #1283.